rest_framework_elasticsearch/es_filters.py

```diff
@@ -45,25 +50,34 @@ def get_schema_fields(self, view):
 
 class ElasticOrderingFilter(filters.OrderingFilter, BaseEsFilterBackend):
 
-    def get_default_ordering(self, view):
-        ordering = getattr(view, 'es_ordering', None)
+    def get_es_ordering_fields(self, view):
+        ordering = view.get_es_ordering_fields() or self.ordering_fields
         if isinstance(ordering, six.string_types):
             return (ordering,)
         return ordering
 
-    def get_valid_fields(self, queryset, view, context={}):
-        valid_fields = getattr(view, 'es_ordering_fields', self.ordering_fields)
-
-        if valid_fields is None:
-            # Default to allowing filtering on serializer fields
-            return self.get_default_valid_fields(queryset, view, context)
+    @staticmethod
+    def validation(field):
+        return (field, field) if isinstance(field, six.string_types) else field
 
-        else:
-            valid_fields = [
-                (item, item) if isinstance(item, six.string_types) else item
-                for item in valid_fields
-                ]
-        return valid_fields
+    def get_valid_fields(self, queryset, view, context={}):
+        fields = self.get_es_ordering_fields(view)
+        if not fields:
+            return tuple()
```

Comment:
By default we allow filtering on serializer fields, that is why we need the context value
`return self.get_default_valid_fields(queryset, view, context)`